builder/proxmox/iso/config.go

```diff
@@ -68,6 +78,90 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, []string, error) {
 		errs = packersdk.MultiErrorAppend(errs, errors.New("when specifying iso_url, iso_storage_pool must also be specified"))
 	}
 
+	// each device type has a maximum number of devices that can be attached.
+	// count iso, disks, additional isos configured for each device type, error if too many.
+	ideCount := 0
+	sataCount := 0
+	scsiCount := 0
+	virtIOCount := 0
+
+	if c.ISODevice == "" {
+		// set default ISO boot device ide2 if no value specified
+		log.Printf("iso_device not set, using default 'ide2'")
+		c.ISODevice = "ide2"
+		ideCount++
+		// Pass c.ISODevice value over to common config to ensure device index not used by disks
+		c.ISOBuilderCDROMDevice = "ide2"
+	} else {
+		// Pass c.ISODevice value over to common config to ensure device index not used by disks
+		c.ISOBuilderCDROMDevice = c.ISODevice
+		// get device from ISODevice config
+		rd := regexp.MustCompile(`\D+`)
+		device := rd.FindString(c.ISODevice)
+		// get index from ISODevice config
+		rb := regexp.MustCompile(`\d+`)
+		_, err := strconv.Atoi(rb.FindString(c.ISODevice))
+		if err != nil {
+			errs = packersdk.MultiErrorAppend(errs, errors.New("iso_device value doesn't contain a valid index number. Expected format is <device type><index number>, eg. scsi0. received value: "+c.ISODevice))
+		}
+		// count iso
+		switch device {
+		case "ide":
+			ideCount++
+		case "sata":
+			sataCount++
+		case "scsi":
+			scsiCount++
+		}
+		for idx, iso := range c.AdditionalISOFiles {
+			if iso.Device == c.ISODevice {
+				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("conflicting device assignment between iso_device (%s) and additional_iso_files block %d device", c.ISODevice, idx+1))
+			}
+			// count additional isos
+			// get device from iso.Device
+			rd := regexp.MustCompile(`\D+`)
+			device := rd.FindString(iso.Device)
+			switch device {
+			case "ide":
+				ideCount++
+			case "sata":
+				sataCount++
+			case "scsi":
+				scsiCount++
+			}
+		}
+		if !(device == "ide" || device == "sata" || device == "scsi") {
+			errs = packersdk.MultiErrorAppend(errs, errors.New("iso_device must be of type ide, sata or scsi. VirtIO not supported for ISO devices"))
```

Comment:
This could be the `default` for the `switch device` above I think?

Reply:
I've updated this to be the `default` for `switch device` in the reroll commit